### DISCUSSION TITLE
posts/2023/04/better-code-image-processing-hugo/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#32] [Announcements]
# posts/2023/04/better-code-image-processing-hugo/

My earlier hack worked, but produced some nasty HTML. Fortunately, two other Hugo users had a better idea.

https://www.brycewray.com/posts/2023/04/better-code-image-processing-hugo/

**Comment by gursuj** [comment]
If you want to clean up the code, you could use the [default](https://gohugo.io/functions/default/) function for setting default parameter values like so:

`{{ $holder := default "GIP" (.Get "holder") }}`

**Reply by brycewray** [reply]
@dinesh-58 Thanks very much! Have made appropriate edits to the real code as well as the relevant code block in this post and its counterparts in subsequent posts, **and** credited you in a footnote in each such post. I don’t know how I’ve missed seeing the `default` documentation up to now, but it surely is a more elegant way, and I appreciate your bringing it to my attention!

**Comment by gursuj** [comment]
Meant to ask about this yesterday. Using `$divClass` while initializing the value for `$divClass` in `imgh.html` throws an error for me. Deleted that bit & it works fine. I inspected the cat image's div & it has a class of bigImgDiv where $divClass was used. Could you explain this?

**Reply by brycewray** [reply]
@dinesh-58 Not sure if this answers your question, but I’ve since moved on to a [unified (Hugo/Cloudinary) shortcode](https://github.com/brycewray/hugo-site/blob/main/layouts/shortcodes/img.html); you might want to read about it [here](https://www.brycewray.com/posts/2023/07/hugo-pipes-cloudinary/) and [here](https://www.brycewray.com/posts/2023/07/hugo-pipes-cloudinary-follow-up/).

(The `bigImgDiv` class is for wider views with most images.)

**Reply by gursuj** [reply]
ah ok, that clarifies where the `bigImgDiv` class came from. I see that, in the Cloudinary blog post, you're setting an initial value for `$divClass` and then updating it & appending the previous value inside an if condition.

There's similar code on line 9 of the `imgh.html` snippet in this post , but this throws an error because `$divClass` value hasn't been set previously. I think you should either set a value for `divClass` before that, or remove it from the `print` function

**Reply by brycewray** [reply]
Ah, yes, I see now. As usual, I was excerpting from what at that time was the real shortcode — which was more complicated because it included options only I would use — and so, in the process of simplifying it for the post’s code example, I apparently left out the initialization of `$divClass`. Sorry! Now fixed. Thanks for the catch!
